### ISSUE TITLE
Spawning too many Python processes, freezes system

### ISSUE THREAD

**Issue by vsalvino** [vscode-restructuredtext/vscode-restructuredtext#205]
**Environment**
- [ ] Extension version: 122.0.0
- [ ] Visual Studio Code version: 1.43.2
- [ ] Operating System version: Windows 10
- [ ] Python version: 3.8.1, 3.8.2
- [ ] sphinx version: latest
- [ ] A sample of conf.py or its important fragments:

```python
# Configuration file for the Sphinx documentation builder.
#
# This file only contains a selection of the most common options. For a full
# list see the documentation:
# https://www.sphinx-doc.org/en/master/usage/configuration.html

# -- Path setup --------------------------------------------------------------

# If extensions (or modules to document with autodoc) are in another directory,
# add these directories to sys.path here. If the directory is relative to the
# documentation root, use os.path.abspath to make it absolute, like shown here.
#
# import sys
# sys.path.insert(0, os.path.abspath("."))
import os
import time

# Absolute path to this directory.
BASE_DIR = os.path.dirname(os.path.abspath(__file__))

# -- Project information -----------------------------------------------------

project = "My Project"
author = "Me"
copyright = f"2019–{time.localtime().tm_year}, {author}"


# -- Options for HTML output -------------------------------------------------

# The theme to use for HTML and HTML Help pages.  See the documentation for
# a list of builtin themes.
# Use our custom theme directory.
html_theme = "theme"

# Search the current directory for theme files.
html_theme_path = [BASE_DIR]

# Add any paths that contain custom static files (such as style sheets) here,
# relative to this directory. They are copied after the builtin static files,
# so a file named "default.css" will overwrite the builtin "default.css".
html_static_path = [
    os.path.join(BASE_DIR, "static"),
]

html_sidebars = {
    "**": ["globaltoc.html"],
}

html_show_sphinx = False

# Print date in "Month D, YYYY" without leading zeros.
html_last_updated_fmt = "%B %-d, %Y"

html_title = project


# -- General configuration ---------------------------------------------------

# Add any Sphinx extension module names here, as strings. They can be
# extensions coming with Sphinx (named "sphinx.ext.*") or your custom
# ones.
extensions = []

# Add any paths that contain templates here, relative to this directory.
# templates_path = [os.path.join(BASE_DIR, "templates")]

# List of patterns, relative to source directory, that match files and
# directories to ignore when looking for source files.
# This pattern also affects html_static_path and html_extra_path.
exclude_patterns = [
    "_build", "Thumbs.db", ".DS_Store", "src",
    # Ignore sass files or dirs
    "sass", "scss", "*.sass", "*.scss",
]
```

**Expected behavior**
Sphinx extension will only run when editing/saving a  `.rst` file, and will not spawn so many python processes.

**Actual behavior**
When saving *any* file (including VS code settings, json, html, etc.), VS Code spwans a ton of python processes which immediately lock up the entire system for a few seconds. The project contains a docs directory with a very small boilerplate sphinx setup. Unfortunately I cannot get a screenshot of the processes, count them, or inspect process info, because when this happens it freezes the entire system until the processes exit.

**Detailed Steps to reproduce the problem**
I have traced the problem down to this sphinx extension, by disabling all others.

1. Install the extension, and the official microsoft python extension. Restart VS Code
2. Open a project/venv/etc. containing a sphinx setup. Be sure to set the python venv in VS code.
3. Set the following extension setting:
```
"python.pythonPath": "C:\\path\\to\\venv\\Scripts\\python.exe",
"restructuredtext.updateOnTextChanged": "false",
"restructuredtext.linter.run": "onSave",
"restructuredtext.confPath": "${workspaceFolder}\\docs"
```
4. Edit any file in the current editor windo, and watch a huge number of python processes spawn when saving the file.

**Additional**

First of all, thank you for this extension. Feature-wise it works perfectly. However it is unusable on my system with the current issue. For the time being I am manually enabling and disabling when editing sphinx docs. I realize this is a difficult issue to track down, but hoping you might have an idea. Will be happy to help however I can!

**Comment by daeki88** [comment]
I have the same problem.

**Comment by hktkzyx** [comment]
I have the same problem

**Comment by andyn** [comment]
As much as I hate shouting "Me too!" this needs to be fixed ASAP.

Not because I'd use this, but because the easiest way to suppress VS Code's warnings about missing .rst file type handlers is to let the IDE semiautomatically install this extension from the marketplace.

**Comment by vsalvino** [comment]
Thanks, will give it a go! Is there any way to limit the number of python processes it spwans? There seems to be dozens just to check one file, one or two could probably suffice (or maybe something like half number of CPUs)

**Comment by lextm** [comment]
Fixed via 359d19693d30e99323f04ab0fd575c780d7f410f

**Comment by vsalvino** [comment]
Seems to have fixed it for me. Thank you!
